api/DeviceAPI.cs

```diff
@@ -89,16 +99,24 @@ public Device registerDevice(string name, UInt64 house_id, string info, UInt64 r
 	 */
 	public bool deleteDevice(Device dev)
 	{
-		if (dev == null)
-            		return false;
-	        //Get[] device list from server API
-	        var dlist = new List<Device>();
-	        //Get[] device list from server API
-	        var item = dlist.First(x => x == dev);
-	        dlist.Remove(item);
+        if (dev == null)
+            return false;
+
+        var dlist = new List<Device>();
+        //dlist = Get[] device list from server API
+        var item = dlist.First(x => x == dev);
+        dlist.Remove(item);
 		return true;
 	}
 
+	public bool Equals(Device x, Device y)
+	{
+		if (object.ReferenceEquals(x, y)) return true;
+
+		if (object.ReferenceEquals(x, null) || object.ReferenceEquals(y, null)) return false;
+
+		return x.ID.HouseID == y.ID.HouseID;
```

Comment:
This comparison seems incomplete, since devices are not uniquely identified by the HouseID.